src/app.php

```diff
@@ -143,10 +147,6 @@ static function (Exception $e, Request $request, $code, GetResponseForExceptionE
                     continue;
                 }
 
-                if (method_exists($event, 'allowCustomResponseCode')) {
-                    $event->allowCustomResponseCode();
-                }
-
                 return $expectation['response'];
```

Comment:
If I understand correctly, I think this line should be moved below, near L155 (which would obviously require to store this in another variable), otherwise we may not run L142 and L143 (ie: increasing `runs`) for all eligible expectations.
But perhaps the "usort" you added in the `MockBuilder` makes it not needed? (anyway, if all tests are green, it probably means I'm wrong)

Reply:
... or we don't have enough tests. Anyway, I'll check this again. Thank you for the review

Reply:
It turns out that you are right: at least a test was missing: here is one which is failing: https://github.com/InterNations/http-mock/pull/62/commits/2a98622ac3d7e57dfb38072a260c64084313fc72
Unfortunately it also turns out that I'm not able to make it green without breaking something else. Most of my attempts end up counting too many runs for some "->twice()" expectations. I'm not sure how to handle that without reintroducing some sort of clumsy `countARunEvenIfLimiterDoesntMatch` :-/

Reply:
Great! Now that we have test to repo the issue it's going to be find to fix it

Reply:
Fixed now. Factoring the consecutive position into the priority seems fine.